packages/core-data/src/batch/test/create-batch.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Internal dependencies
+ */
+import createBatch from '../create-batch';
+
+describe( 'createBatch', () => {
+	test( 'running an empty batch', async () => {
+		const processor = async ( inputs ) => inputs;
+		const batch = createBatch( processor );
+		expect( await batch.run() ).toBe( true );
+	} );
+
+	test( 'running resolves promises when processor returns output', async () => {
+		const processor = async ( inputs ) =>
+			inputs.map( ( input ) => ( {
+				output: input,
+			} ) );
+		const batch = createBatch( processor );
+		const promise1 = batch.add( 1 );
+		const promise2 = batch.add( 2 );
+		expect( await batch.run() ).toBe( true );
+		expect( await promise1 ).toBe( 1 );
+		expect( await promise2 ).toBe( 2 );
```

Comment:
Maybe we would also want to test if both `promise1` and `promise2` have been fulfilled if `batch.run()` has been fulfilled. Something like:

```js
let promise1Value = null;
let promise2Value = null;
batch.add(1).then((value) => { promise1Value = value });
batch.add(2).then((value) => { promise2Value = value });
expect(await batch.run()).toBe(true);
expect(promise1Value).toBe(1):
expect(promise2Value).toBe(2):
```

Reply:
Doesn't this test the same thing? Not sure I understand.

Reply:
I don't think the original way tests that the promises would fulfilled once `batch` has finished.

`promise1` and `promise2` could still be pending after `batch.run()` has been fulfilled.

Reply:
Ah! Gotcha. That's probably what I want to do for all tests.

Reply:
34db07a19b